src/log/log.c

```diff
@@ -1166,11 +1166,11 @@ __log_newfile(WT_SESSION_IMPL *session, bool conn_open, bool *created)
      * set, force that ordering.
      */
     if (log->log_fh == NULL)
-        log->log_close_fh = NULL;
+        __wt_atomic_store_pointer(&log->log_close_fh, NULL);
     else {
         WT_ASSIGN_LSN(&log->log_close_lsn, &log->alloc_lsn);
         /* Paired with an acquire read in the log file server path. */
-        WT_RELEASE_WRITE_WITH_BARRIER(log->log_close_fh, log->log_fh);
```

Comment:
To be clear this is a weakening of ordering. When we added the new macros `WT_AQUIRE_READ` and `WT_RELEASE_WRITE` we only intended to use them for _new_ usages and not go around replacing past usages. However I would like to see them eventually replace the older barriers which leak ordering semantics. 

How convinced are you that the barrier, which is a stronger semantics, doesn't have a beneficial side effect? I don't have time to review the algos here in detail.

Reply:
I see, I saw the deprecated comment and thought that we want to slowly replace older barriers to new barriers. I want to avoid any potential algorthmic and concurrency discussion in this PR and focus on fixing the TSAN warnings. I will move them back.